### PR TITLE
Record tracks on product edit/update

### LINKED ISSUE #37656: Product Block Editor: Add save draft and publish product tracking
**Description**

We should track when a user has saved the product as either draft or published the product within the new product block editor.
We already have two events for this:
https://mc.a8c.com/tracks/events/event.php?eventname=wcadmin_product_edit ( triggered when saving draft )
https://mc.a8c.com/tracks/events/event.php?eventname=wcadmin_product_update (triggered when publishing or saving a published product )

We should include the `new_product_page` property and set it to `true` for the Product block editor.
You can see our last implementation: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/products/product-form-actions.tsx#L78

**Acceptance criteria**

- [ ] `wcadmin_product_edit` should be fired with `new_product_page` prop set to true when user saves product as draft within product block editor
- [ ] `wcadmin_product_update` should be fired with `new_product_page` prop set to true when user saves product as publish within product block editor

### PR DESCRIPTION
**Pull request by joshuatf**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Adds the tracks events when saving a draft or saving/publishing a product.

Screen Shot 2023-04-17 at 4 46 34 PM
Screen Shot 2023-04-17 at 4 24 54 PM

Closes #37656  .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1.. Navigate to Tools -> WCA Test Helper -> Features and enable the new product blocks editing experience
2. Navigate to Products -> Add new
3. Open your console
4. Run `localStorage.setItem( 'debug', 'wc-admin:tracks' );` in the console.
5. Make some edits to the product
6. Click "Save draft"
7. Note that the `wcadmin_product_edit` and its properties are shown in the console
8. Make some edits to the product
9. Click "Publish" or "Save"
10. Note that the `wcadmin_product_update` and its properties are shown in the console

### CODE REVIEW

File: packages/js/data/src/products/types.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { BaseQueryParams } from '../types';
 
 export type ProductType = 'simple' | 'grouped' | 'external' | 'variable';
 export type ProductStatus =
+	| 'auto-draft'
```
Comment (mdperez86) [inline packages/js/data/src/products/types.ts:14]: 🙌

File: packages/js/product-editor/src/components/header/publish-button/publish-button.tsx [hunk 2]
```diff
@@ -24,14 +25,17 @@ export function PublishButton(
 		'status'
 	);
 
-	const isCreating = productStatus === 'auto-draft';
+
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/header/publish-button/publish-button.tsx]: Extra line here

File: packages/js/data/src/products/types.ts (hunk 1 above)
Comment (mdperez86) [reply packages/js/data/src/products/types.ts:14]: After adding this new status here this [line](https://github.com/woocommerce/woocommerce/pull/37770/files#diff-4004a9c9836f76eca626ac6f089db52eb4cc3b93724c4a2186ae0b4c81844870R22) now should looks like `const [ productStatus ] = useEntityProp< ProductStatus >(` right. I know this is not part of this PR but you touched the `ProductStatus` type also.

File: packages/js/product-editor/src/components/header/publish-button/publish-button.tsx [hunk 3]
```diff
@@ -24,14 +25,17 @@ export function PublishButton(
...
+	const isCreating = status === 'auto-draft';
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/header/publish-button/publish-button.tsx]: Where `status` variable is coming from?

**Review by mdperez86 — CHANGES_REQUESTED** [review]
Goo job @joshuatf !! I just left some small comments.

File: packages/js/data/src/products/types.ts (hunk 1 above)
Comment (joshuatf) [reply packages/js/data/src/products/types.ts:14]: Good call!  The above link did not work for me, but I assumed it was everywhere with `ProductStatus | 'auto-draft'`.  Replaced those in https://github.com/woocommerce/woocommerce/pull/37770/commits/7c44bb773ffcc14d00af8c3078d8695660a3e600

File: packages/js/product-editor/src/components/header/publish-button/publish-button.tsx (hunk 3 above)
Comment (joshuatf) [reply packages/js/product-editor/src/components/header/publish-button/publish-button.tsx]: Accidentally left over from the previous iteration before I moved the tracking utils out.  So weird my compiler didn't complain.  Fixed!

**Review by mdperez86 — APPROVED** [review]
Nice job @joshuatf LGTM!!!

**Pull request merged**